File: python/copass-management/src/copass_management/tools/get_agent.py
from __future__ import annotations

from typing import Any, Dict, TYPE_CHECKING

if TYPE_CHECKING:
    from copass_management.registrar import ToolContext


async def get_agent(ctx: "ToolContext", input: Dict[str, Any]) -> Any:
    return await ctx.client.agents.retrieve(ctx.sandbox_id, str(input["slug"]))

File: typescript/packages/management/src/tools/get_agent.ts
import type { ToolContext, ToolHandler } from '../registrar.js';

export const getAgent: ToolHandler = async (
  ctx: ToolContext,
  input: Record<string, unknown>,
) => {
  const slug = String(input.slug);
  return ctx.client.agents.retrieve(ctx.sandboxId, slug);
};
